### PR TITLE
Relax primal-dual system upon linesearch failure

### LINKED ISSUE #7: Terminate earlier on failures?
A slightly modified version of the toy example in the README currently causes many linesearch-failures before the solver gives up. The underpinning problem seems to be that the inner loop breaks early and then the outer-loop concludes that the sub-problem was easy to solve, causing it to tighten `\epsilon` further.

Reproducer:

```julia
using MixedComplementarityProblems

M = [0 1; -1 0]
A = [1 0; 0 1]
b = [1; 1]
θ = rand(2)

G(x, y; θ) = M * x - θ - A' * y
H(x, y; θ) = A * x - b

mcp = MixedComplementarityProblems.PrimalDualMCP(
    G,
    H;
    unconstrained_dimension = size(M, 1),
    constrained_dimension = length(b),
    parameter_dimension = size(M, 1),
)
sol = MixedComplementarityProblems.solve(MixedComplementarityProblems.InteriorPoint(), mcp, θ)
```

### PR DESCRIPTION
**Pull request by dfridovi**
Adjusts `\epsilon` as follows:

- if linesearch succeeds: `\epsilon *= 1 - exp(-iters)`
- if linesearch fails: `\epsilon *= 1 + exp(-iters)`

This fixes #7.

### CODE REVIEW

**Comment by lassepe** [comment]
I thought I had tested something like this and concluded that it wasn't enough because it cycles. Maybe try my MVP from #7 as a test?

**Comment by dfridovi** [comment]
I think the `runtests.jl` file should have exactly that MVP, no?

**Comment by lassepe** [comment]
M is different: `M = [0 1; -1 0]`

**Comment by dfridovi** [comment]
Right... helps to read ;) So that `M` makes the QP nonconvex, and therefore I think with these constraints it is going to be an ill-posed problem. That explains why it was having trouble. I've added a "fix" that just errors out after a fixed number of outer loop iterations now.

**Comment by dfridovi** [comment]
Ah I see - basically the quadratic term will evaluate to zero for all x. Then the well posedness entirely depends on the sign of theta.

File: test/runtests.jl [hunk 1]
```diff
@@ -14,9 +14,10 @@ using FiniteDiff: FiniteDiff
                                  0 ≤ y ⟂ H(x, y) = Ax - b ≥ 0.
     """
     M = [2 1; 1 2]
+    #    M = [0 1; -1 0]
```
Comment (lassepe) [inline test/runtests.jl:17]: This is never used. Intended?

Comment (dfridovi) [reply test/runtests.jl:17]: Good catch

**Pull request merged**